scheduler/system_util.go

```diff
@@ -22,7 +22,11 @@ func materializeSystemTaskGroups(job *structs.Job) map[string]*structs.TaskGroup
 	}
 
 	for _, tg := range job.TaskGroups {
-		for i := 0; i < tg.Count; i++ {
+		count := tg.Count
+		if job.Type == structs.JobTypeSystem {
+			count = 1 // system jobs don't accept a count
+		}
```

Comment:
seems like this belongs in `structs.TaskGroup.Canonicalize`? 

no problem in having both i suppose, but this is an odd place to be modifying the job

Reply:
I thought it already was! But it turns out it's being canonicalized in `api/tasks.go` which means that the jobs we get from `mock.Job()` haven't been correctly canonicalized for counts. I'll move this fix there.